cscs-checks/prgenv/ssh_environ_check.py

```diff
@@ -0,0 +1,36 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class SSHLoginEnvCheck(rfm.RunOnlyRegressionTest):
+    def __init__(self):
+        super().__init__()
+        self.descr = ('Check the values of a set of environment variables '
+                      'when accessing remotely over SSH')
+        self.valid_systems = ['daint:login', 'dom:login']
+        self.valid_prog_environs = ['PrgEnv-cray']
+        reference = {
+            'CRAY_CPU_TARGET': 'haswell',
+            'CRAYPE_NETWORK_TARGET': 'aries',
+            'MODULEPATH': r'[\S+]',
+            'MODULESHOME': r'/opt/cray/pe/modules/[\d+\.+]',
+            'PE_PRODUCT_LIST': 'CRAYPE_HASWELL:CRAY_RCA:CRAY_ALPS:DVS:'
+                               'CRAY_XPMEM:CRAY_DMAPP:CRAY_PMI:CRAY_UGNI:'
+                               'CRAY_UDREG:CRAY_LIBSCI:CRAYPE:CRAY:'
+                               'PERFTOOLS:CRAYPAT',
+            'SCRATCH': r'/scratch/[\S+]',
+            'XDG_RUNTIME_DIR': r'/run/user/[\d+]'
+        }
+        self.executable = 'ssh'
+        echo_args = ' '.join('%s=$%s' % (i, i)  for i in reference.keys())
+        self.executable_opts = [self.current_system.name,
+                                'echo', "'%s'" % echo_args]
+        self.sanity_patterns = sn.all(sn.map(self.assert_envvar,
+                                      list(reference.keys()),
+                                      list(reference.values())))
```

Comment:
You can rewrite this in a simpler way:

```python
        self.sanity_patterns = sn.all(
            sn.map(self.assert_envvar, list(reference.items())))
```

See below how the `assert_envvar` is also simplified.